api/api_test.go

```diff
@@ -395,6 +399,105 @@ func TestGetTeamName(t *testing.T) {
 	})
 }
 
+func TestGetTeamRoster(t *testing.T) {
+	teamUUID := uuid.Must(uuid.NewV4())
+
+	requesterKey, err := pgpkey.LoadFromArmoredEncryptedPrivateKey(
+		exampledata.ExamplePrivateKey4, "test4",
+	)
+	assert.NoError(t, err)
+
+	expectedRoster := "fake roster"
+	expectedSignature := "fake signature"
+
+	decryptedRosterAndSig := v1structs.TeamRosterAndSignature{
+		TeamRoster:               expectedRoster,
+		ArmoredDetachedSignature: expectedSignature,
+	}
+
+	jsonRosterAndSig, err := json.MarshalIndent(decryptedRosterAndSig, "", "    ")
+	assert.NoError(t, err)
+
+	encryptedJSON, err := encryptToArmor(t, string(jsonRosterAndSig), requesterKey)
+	assert.NoError(t, err)
+
+	client, mux, _, teardown := setup()
+	defer teardown()
+
+	teamRosterResponse, err := json.Marshal(v1structs.GetTeamRosterResponse{
+		EncryptedJSON: encryptedJSON,
+	})
+	assert.NoError(t, err)
+
+	mockResponseHandler := func(w http.ResponseWriter, r *http.Request) {
+		assertClientSentVerb(t, "GET", r.Method)
+		w.Header().Add("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
+		fmt.Fprint(w, string(teamRosterResponse))
+	}
+	mux.HandleFunc(
+		fmt.Sprintf("/team/%s/roster", teamUUID),
+		mockResponseHandler,
+	)
+
+	t.Run("decrypts the roster with a valid requester key", func(t *testing.T) {
+		gotRoster, gotSignature, err := client.GetTeamRoster(*requesterKey, teamUUID)
+
+		assert.NoError(t, err)
+		assert.Equal(t, expectedRoster, gotRoster)
+		assert.Equal(t, expectedSignature, gotSignature)
+	})
+
+	t.Run("returns an error given an invalid requester key", func(t *testing.T) {
+		invalidRequesterKey, err := pgpkey.LoadFromArmoredEncryptedPrivateKey(
+			exampledata.ExamplePrivateKey2, "test2",
+		)
+		assert.NoError(t, err)
+
+		_, _, err = client.GetTeamRoster(*invalidRequesterKey, teamUUID)
+
+		assert.GotError(t, err)
+		assert.Equal(t,
+			fmt.Errorf("couldn't decrypt json: error reading message: openpgp: incorrect key"), err,
+		)
+	})
+
+	t.Run("404 returns a specific type of error", func(t *testing.T) {
+		unknownUUID := uuid.Must(uuid.NewV4())
+		mockNotFoundResponseHandler := func(w http.ResponseWriter, r *http.Request) {
+			assertClientSentVerb(t, "GET", r.Method)
+			w.Header().Add("Content-Type", "application/json")
+			w.WriteHeader(http.StatusNotFound)
+		}
+		mux.HandleFunc(
+			fmt.Sprintf("/team/%s/roster", unknownUUID),
+			mockNotFoundResponseHandler,
+		)
+
+		_, _, err := client.GetTeamRoster(*requesterKey, unknownUUID)
+
+		assert.Equal(t, ErrTeamNotFound, err)
+	})
+
+	t.Run("responds with http 500 (unexpected http code)", func(t *testing.T) {
+		errorUUID := uuid.Must(uuid.NewV4())
+		mockErrorResponseHandler := func(w http.ResponseWriter, r *http.Request) {
+			assertClientSentVerb(t, "GET", r.Method)
```

Comment:
I reckon another one of these test methods would be helpful for the future: `assertClientSentValidAuthHeader`